Makefile

```diff
@@ -1,3 +1,3 @@
 all: src/*.cpp inc/*.hpp
-	g++ -std=c++17 src/*.cpp -o main
+	g++ -std=c++17 -Werror  src/*.cpp -o main
```

Comment:
Not only `-Werror` should be added. Please add `-Wall -Wextra -pedantic -Wconversion`